splunk-otel-android/src/test/java/com/splunk/rum/SpanFilterTest.java

```diff
@@ -0,0 +1,226 @@
+                .build()
+                .apply(delegate);
+
+        SpanData rejected = span("span", Attributes.of(ATTRIBUTE, "test"));
+        SpanData differentKey = span("span", Attributes.of(OTHER_ATTRIBUTE, "test", LONG_ATTRIBUTE, 42L));
+        SpanData anotherRejected = span("span", Attributes.of(ATTRIBUTE, "rejected!"));
+        SpanData differentValue = span("span", Attributes.of(ATTRIBUTE, "not really test"));
+        SpanData yetAnotherRejected = span("span", Attributes.of(ATTRIBUTE, "pass", LONG_ATTRIBUTE, 123L));
+
+        CompletableResultCode expectedResult = new CompletableResultCode();
+        when(delegate.export(asList(differentKey, differentValue))).thenReturn(expectedResult);
+
+        // when
+        CompletableResultCode result = underTest.export(
+                asList(rejected, differentKey, anotherRejected, differentValue, yetAnotherRejected));
+
+        // then
+        assertSame(expectedResult, result);
+    }
+
+    @Test
+    public void shouldRemoveSpanAttributes() {
+        // given
+        SpanExporter underTest = new SpanFilterBuilder()
+                .removeSpanAttribute(ATTRIBUTE, value -> value.equals("test"))
+                // make sure that attribute types are taken into account
+                .removeSpanAttribute(stringKey("long_attribute"))
+                .build()
+                .apply(delegate);
+
+        SpanData span1 = span("first", Attributes.of(ATTRIBUTE, "test", LONG_ATTRIBUTE, 42L));
+        SpanData span2 = span("second", Attributes.of(ATTRIBUTE, "not test", OTHER_ATTRIBUTE, "test"));
+
+        CompletableResultCode expectedResult = new CompletableResultCode();
+        when(delegate.export(spansCaptor.capture())).thenReturn(expectedResult);
+
+        // when
+        CompletableResultCode result = underTest.export(asList(span1, span2));
+
+        // then
+        assertSame(expectedResult, result);
+
+        List<SpanData> exportedSpans = new ArrayList<>(spansCaptor.getValue());
+        assertEquals(2, exportedSpans.size());
+        assertEquals("first", exportedSpans.get(0).getName());
+        assertEquals(Attributes.of(LONG_ATTRIBUTE, 42L), exportedSpans.get(0).getAttributes());
+        assertEquals("second", exportedSpans.get(1).getName());
+        assertEquals(Attributes.of(ATTRIBUTE, "not test", OTHER_ATTRIBUTE, "test"), exportedSpans.get(1).getAttributes());
+    }
+
+    @Test
+    public void shouldReplaceSpanAttributes() {
+        // given
+        SpanExporter underTest = new SpanFilterBuilder()
+                .replaceSpanAttribute(ATTRIBUTE, value -> value + "!!!")
+                .replaceSpanAttribute(ATTRIBUTE, value -> value + "1")
+                .replaceSpanAttribute(LONG_ATTRIBUTE, value -> value + 1)
+                // make sure that attribute types are taken into account
+                .replaceSpanAttribute(stringKey("long_attribute"), value -> "abc")
```

Comment:
I've actually wondered if this might end up being more confusing than simply matching on the key's string value. I honestly don't know if users will grok this difference, but we can always loosen it up later if we get complaints or confusion.

Reply:
We can probably add overloads (like all those `AttributesBuilder#put()` methods) if needed.